File: backend/src/main/java/com/slior/dto/route/OptimizeRouteRequest.java
package com.slior.dto.route;

import com.slior.model.enums.VehicleType;
import jakarta.validation.constraints.NotNull;

public record OptimizeRouteRequest(

        @NotNull(message = "La latitud del punto de inicio es obligatoria")
        Double puntoInicioLat,

        @NotNull(message = "La longitud del punto de inicio es obligatoria")
        Double puntoInicioLon,

        @NotNull(message = "El tipo de vehículo es obligatorio")
        VehicleType vehicleType
) {}

File: mobile-app/app/src/main/java/com/slior/data/remote/dto/OptimizeRouteRequest.kt
package com.slior.data.remote.dto

/**
 * DTO para la solicitud de optimización de ruta.
 * Incluye la ubicación del inicio y el tipo de vehículo para optimizar según velocidad.
 */
data class OptimizeRouteRequest(
    val puntoInicioLat: Double,
    val puntoInicioLon: Double,
    val vehicleType: String
)
